yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/compose/Composer.scala

```diff
@@ -18,7 +19,8 @@ trait Composer:
   def fromEvents(events: List[Event]): Either[YamlError, Node]
 
 object ComposerImpl extends Composer:
-  type ComposeResult[T] = Either[YamlError, (T, List[Event])]
+  type ComposeResult[T]        = Either[YamlError, (T, List[Event])]
+  type ComposeResultWithPos[T] = Either[YamlError, (T, Option[Position], List[Event])]
```

Comment:
Why do we need to separate into two types? 

`(T, Option[Position], List[Event])` it can wrapped to new type or case class. It will be more readable

Reply:
Those types should be private. `ComposeResultWithPos` is used in tailrecs because they should return triple (parsed node, first child position, rest of the events). Thanks to that other methods don't bother with unused position parameter